internal/vds/vds.cpp

```diff
@@ -554,6 +569,23 @@ struct vdsbuffer fetch_fence(
 
         auto coordinate = transform_coordinate(x, y);
 
+        auto validate_boundary = [&] (const int voxel) {
```

Comment:
The compiler will optimise this, but maybe still move the lambda outside the loop?